src/Data/AVL/Key.agda

```diff
@@ -18,32 +18,41 @@ open import Data.Empty
 open import Data.Unit
 open import Data.Product
 open import Relation.Binary.PropositionalEquality using (_≡_ ; refl)
+open import Relation.Nullary.Construct.Add.Extrema
+  as AddExtremaToSet using (_±)
+open import Relation.Binary.Construct.Add.Extrema.Strict
+  as AddExtremaToOrder using ()
 
 open StrictTotalOrder strictTotalOrder renaming (Carrier to Key)
 
 -----------------------------------------------------------------------
--- Definition
+-- Keys are augmented with new extrema (i.e. an artificial minimum and
+-- maximum)
 
-infix 5 [_]
+Key⁺ : Set a
+Key⁺ = Key ±
 
-data Key⁺ : Set a where
-  ⊥⁺ ⊤⁺ : Key⁺
-  [_]   : (k : Key) → Key⁺
+open AddExtremaToSet public
+  using ([_])
+  renaming
+  ( ⊥± to ⊥⁺
+  ; ⊤± to ⊤⁺
+  )
 
-[_]-injective : ∀ {k l} → [ k ] ≡ [ l ] → k ≡ l
+[_]-injective : ∀ {k l : Key} → [ k ] ≡ [ l ] → k ≡ l
 [_]-injective refl = refl
 
```

Comment:
Is this a renamed [`[≈]-injective`](https://agda.github.io/agda-stdlib/Relation.Binary.Construct.Add.Extrema.Equality.html#1195)?

Reply:
Answering my own question: no. `[≈]-injective` is for pointwise equality.
We should maybe have `[_]-injective` directly in the extrema lib?